homeassistant/components/tuya/__init__.py

```diff
@@ -296,3 +302,30 @@ def async_remove_device(self, device_id: str) -> None:
         if device_entry is not None:
             device_registry.async_remove_device(device_entry.id)
             self.device_ids.discard(device_id)
+
+
+class MessageListener:
```

Comment:
This can be part of `DeviceListener`? We don't need separate handling for this?

Reply:
Hi,

Unfortunatelly no - to handle scene switches we don't want to have state of device - we need individual events. Device state won't handle situation when button is pressed multiple times - from DeviceListener point of view (and later HA's) state does not change - because button state is "pressed" the whole time.